agera/src/main/java/com/google/android/agera/BaseObservable.java

```diff
@@ -131,29 +143,31 @@ private void remove(@NonNull final Updatable updatable) {
     throw new IllegalStateException("Updatable not added, cannot remove.");
   }
 
-  synchronized void sendUpdate() {
-    handler.removeMessages(WorkerHandler.MSG_UPDATE, this);
-    final long elapsedRealtimeMillis =
-        shortestUpdateWindowMillis > 0 ? elapsedRealtime() : 0;
-    final long timeFromLastUpdate = elapsedRealtimeMillis - lastUpdateTimestamp;
-    if (timeFromLastUpdate >= shortestUpdateWindowMillis) {
-      lastUpdateTimestamp = elapsedRealtimeMillis;
+  void sendUpdate() {
+    synchronized (lock) {
+      if (shortestUpdateWindowMillis > 0) {
+        final long elapsedRealtimeMillis = elapsedRealtime();
+        final long timeFromLastUpdate = elapsedRealtimeMillis - lastUpdateTimestamp;
+        if (timeFromLastUpdate  < shortestUpdateWindowMillis) {
+          handler.sendMessageDelayed(handler.obtainMessage(WorkerHandler.MSG_UPDATE, this),
+              shortestUpdateWindowMillis - timeFromLastUpdate);
+          pendingUpdate = true;
```

Comment:
This should be a precondition to the whole method (if !pendingUpdate return). After all, you're setting pendingUpdate to true in dispatchUpdate(), i.e. before the call to sendUpdate() is scheduled.